src/databricks/labs/remorph/reconcile/query_builder_refactored/expression_generator.py

```diff
@@ -0,0 +1,159 @@
+from sqlglot.expressions import (
+    Alias,
+    Anonymous,
+    Boolean,
+    Coalesce,
+    Column,
+    DataType,
+    Expression,
+    From,
+    Join,
+    JSONFormat,
+    Literal,
+    Trim,
+)
+
+
+def _apply_func_expr(expr: Expression, expr_func: Callable, **kwargs) -> Expression:
+    level = 0 if isinstance(expr, exp.Column) else 1
+    new_expr = expr.copy()
+    for node in new_expr.dfs():
+        if isinstance(node, exp.Column):
+            column_name = node.name
+            table_name = node.table
+            func = expr_func(this=exp.Column(this=column_name, table=table_name), **kwargs)
+            if level == 0:
+                return func
+            node.replace(func)
+    return new_expr
+
+
+def coalesce(expr: Expression, default="0", is_string=False) -> Coalesce | Expression:
+    expressions = [exp.Literal(this=default, is_string=is_string)]
+    return _apply_func_expr(expr, exp.Coalesce, expressions=expressions)
+
+
+def trim(expr: Expression) -> Trim | Expression:
+    return _apply_func_expr(expr, exp.Trim)
+
+
+def json_format(expr: Expression, options: dict[str, str] | None = None) -> JSONFormat | Expression:
+    return _apply_func_expr(expr, exp.JSONFormat, options=options)
+
+
+def sort_array(expr: Expression, asc=True):
+    return _apply_func_expr(expr, exp.SortArray, asc=Boolean(this=asc))
+
+
+def to_char(expr: Expression, to_format=None, nls_param=None):
+    if to_format:
+        return _apply_func_expr(
+            expr, exp.ToChar, format=exp.Literal(this=to_format, is_string=True), nls_param=nls_param
+        )
+    return _apply_func_expr(expr, exp.ToChar)
+
+
+def array_to_string(
+    expr: Expression, delimiter: str = ",", is_string=True, null_replacement: str | None = None, is_null_replace=True
+):
+    if null_replacement:
+        return _apply_func_expr(
+            expr,
+            exp.ArrayToString,
+            expression=[exp.Literal(this=delimiter, is_string=is_string)],
+            null=exp.Literal(this=null_replacement, is_string=is_null_replace),
+        )
+    return _apply_func_expr(expr, exp.ArrayToString, expression=[exp.Literal(this=delimiter, is_string=is_string)])
+
+
+def array_sort(expr: Expression, asc=True):
+    return _apply_func_expr(expr, exp.ArraySort, expression=Boolean(this=asc))
+
+
+def anonymous(expr: Expression, func: str) -> Anonymous | Expression:
+    """
+
+    This function used in cases where the sql functions are not available in sqlGlot expressions
+    Example:
+        >>> from sqlglot import parse_one
+        >>> print(repr(parse_one('select unix_timestamp(col1)')))
+
+    the above code gives you a Select Expression of Anonymous function.
+
+    To achieve the same,we can use the function as below:
+    eg:
+        >>> expr = parse_one("select col1 from dual")
+        >>> transformed_expr=anonymous(expr,"unix_timestamp")
+        >>> print(transformed_expr)
+        'SELECT UNIX_TIMESTAMP(col1) FROM DUAL'
```

Comment:
This UNIX_TIMESTAMP needs to move to `local_expression` Please create a issue as tech debt.

Reply:
Created an issue.#272